src/store/product-store.ts

```diff
@@ -0,0 +1,41 @@
+import { makeAutoObservable, runInAction } from 'mobx';
+import { messages } from '../sources/messages';
+import { AxiosError } from 'axios';
+import { productService } from '../api/services/product-service.ts';
+import type { Product } from '../sources/types/product';
+
+export class ProductStore {
+  public product: Product.Product | null;
+  public isLoading = false;
+  public error: string | null = null;
+
+  constructor() {
+    makeAutoObservable(this);
+    this.product = null;
+  }
+
+  public getProduct = async (id: string) => {
+    this.isLoading = true;
+    this.error = null;
+    this.product = null;
+    try {
+      const data = await productService.getProductByID(id);
+      runInAction(() => {
+        this.product = data;
```

Comment:
```suggestion
        this.product = prepareProduct(data);
```
давай преобразуем в понятный вид объект, перед тем как в стор записать. так будет проще с ним работать в дальнейшем